Knossos.NET/Models/Mod.cs

```diff
@@ -348,127 +348,221 @@ public List<string> GetModFlagList(bool overrideSettings = false)
-                                        temp.Remove(eqOrHigh);
-                                    }
-                                });
-                                revisions.ForEach(otherrev =>
+                                    lower.Remove(l);
+                                }
+                            });
+                            //Remove any "higher" that can be sastified by this rev
+                            higher.ToList().ForEach(h =>
+                            {
+                                if (SemanticVersion.SastifiesDependency(h.version, re.version!.Replace("~", "")))
                                 {
-                                    if (otherrev != revs && SemanticVersion.SastifiesDependency(revs.version, otherrev.version!.Replace("~", "")))
-                                    {
-                                        temp.Remove(otherrev);
-                                    }
-                                });
-                            }
-                            //Equal or Higher at this point if anything left it can only be other equal or higher, remove the ones that are not incluided on others
-                            foreach (var eqOrHigher in equalOrHigher)
+                                    higher.Remove(h);
+                                }
+                            });
+                        });
+
+                        //<=
+                        lower.ToList().ForEach(lo => {
+                            //Remove any "lower" that can be sastified by this other lower
+                            lower.ToList().ForEach(l =>
                             {
-                                equalOrHigher.ForEach(otherEqOrHigher =>
+                                if (lo != l && SemanticVersion.SastifiesDependency(l.version, lo.version!.Replace("<=", "")))
                                 {
-                                    if (otherEqOrHigher != eqOrHigher && SemanticVersion.SastifiesDependency(eqOrHigher.version, otherEqOrHigher.version!.Replace(">=", "")))
-                                    {
-                                        temp.Remove(eqOrHigher);
-                                    }
-                                });
-                            }
+                                    lower.Remove(l);
+                                }
+                            });
+                        });
+                        lower.ForEach(lo => {
+                            //Remove any "higher" that can be sastified by this lower
+                            higher.ToList().ForEach(h =>
+                            {
+                                if (SemanticVersion.SastifiesDependency(h.version, lo.version!.Replace("<=", "")))
+                                {
+                                    higher.Remove(h);
+                                }
+                            });
+                        });
```

Comment:
Hmmmm ... the way I'm reading this, if `l` is `<=1.1.0` and `h` is `>=1.0.2`, `h` is filtered out.  But that allows us to use `1.0.1`.  Is that desired?

Reply:
Yes thats the flaw with the current dependency system